File: solutions/csharp/line-up/1/LineUp.cs
using System.Runtime.InteropServices.ComTypes;

public static class LineUp
{
    public static string Format(string name, int number) =>
        $"{name}, you are the {Formatted(number)} customer we serve today. Thank you!";

    private static string Formatted(int number) =>
        $"{number}{Suffix(number)}";

    private static string Suffix(int number) =>
        (number % 10, number % 100) switch
        {
            (_, 11) => "th",
            (_, 12) => "th",
            (_, 13) => "th",
            (1, _) => "st",
            (2, _) => "nd",
            (3, _) => "rd",
            _ => "th"
        };
}

File: solutions/go/line-up/1/line_up.go
package lineup

import "fmt"

func ordinal(number int) string {
	if number%100 != 11 && number%10 == 1 {
		return fmt.Sprintf("%dst", number)
	}

	if number%100 != 12 && number%10 == 2 {
		return fmt.Sprintf("%dnd", number)
	}

	if number%100 != 13 && number%10 == 3 {
		return fmt.Sprintf("%drd", number)
	}

	return fmt.Sprintf("%dth", number)
}

func Format(name string, number int) string {
	return fmt.Sprintf("%s, you are the %s customer we serve today. Thank you!", name, ordinal(number))
}